diff --git a/core/src/main/java/io/bitsquare/trade/offer/OfferBookService.java b/core/src/main/java/io/bitsquare/trade/offer/OfferBookService.java
index c3650be..f30d4da 100644
--- a/core/src/main/java/io/bitsquare/trade/offer/OfferBookService.java
+++ b/core/src/main/java/io/bitsquare/trade/offer/OfferBookService.java
@@ -17,6 +17,8 @@
 
 package io.bitsquare.trade.offer;
 
+import py4j.GatewayServer;
+
 import io.bitsquare.btc.pricefeed.PriceFeed;
 import io.bitsquare.common.handlers.ErrorMessageHandler;
 import io.bitsquare.common.handlers.ResultHandler;
@@ -79,6 +81,11 @@ public class OfferBookService {
                 });
             }
         });
+
+        // start py4j server
+        GatewayServer gatewayServer = new GatewayServer(this);
+        gatewayServer.start();
+        System.out.println("Gateway Server Started");
     }
 
     public void addOfferBookChangedListener(OfferBookChangedListener offerBookChangedListener) {
diff --git a/pom.xml b/pom.xml
index 27e193f..7e72ddf 100755
--- a/pom.xml
+++ b/pom.xml
@@ -183,5 +183,13 @@
             <version>4.1.1.RELEASE</version>
             <scope>test</scope>
         </dependency>
+
+        <!--py4j-->
+        <!-- http://mvnrepository.com/artifact/net.sf.py4j/py4j -->
+        <dependency>
+            <groupId>net.sf.py4j</groupId>
+            <artifactId>py4j</artifactId>
+            <version>0.10.1</version>
+        </dependency>
     </dependencies>
-</project>
\ No newline at end of file
+</project>
